thanos/templates/compact-deployment.yaml

```diff
@@ -1,108 +1,140 @@
+{{ with  $root.Values.compact.labels }}{{ toYaml . | indent 8 }}{{ end }}
+      {{- if or $root.Values.compact.annotations $root.Values.compact.metrics.annotations.enabled }}
       annotations:
       {{- end }}
-      {{- with  .Values.compact.annotations }}{{ toYaml . | nindent 8 }}{{- end }}
-      {{- if .Values.compact.metrics.annotations.enabled  }}
+      {{- with  $root.Values.compact.annotations }}{{ toYaml . | nindent 8 }}{{- end }}
+      {{- if $root.Values.compact.metrics.annotations.enabled  }}
         prometheus.io/scrape: "true"
-        prometheus.io/port: "{{ .Values.compact.http.port }}"
+        prometheus.io/port: "{{ $root.Values.compact.http.port }}"
       {{- end }}
     spec:
       containers:
       - name: thanos-compact
-        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
-        imagePullPolicy: {{ .Values.image.pullPolicy }}
-        {{- with .Values.compact.extraEnv }}
+        image: "{{ $root.Values.image.repository }}:{{ $root.Values.image.tag }}"
+        imagePullPolicy: {{ $root.Values.image.pullPolicy }}
+        {{- with $root.Values.compact.extraEnv }}
         env: {{ toYaml . | nindent 8 }}
         {{- end }}
         args:
         - "compact"
-        - "--log.level={{ .Values.compact.logLevel }}"
-        - "--log.format={{ .Values.compact.logFormat }}"
-        - "--http-address=0.0.0.0:{{ .Values.compact.http.port }}"
+        - "--log.level={{ $root.Values.compact.logLevel }}"
+        - "--log.format={{ $root.Values.compact.logFormat }}"
+        - "--http-address=0.0.0.0:{{ $root.Values.compact.http.port }}"
         - "--objstore.config-file=/etc/config/object-store.yaml"
         - "--data-dir=/var/thanos/compact"
-        - "--consistency-delay={{ .Values.compact.consistencyDelay }}"
-        - "--retention.resolution-raw={{ .Values.compact.retentionResolutionRaw }}"
-        - "--retention.resolution-5m={{ .Values.compact.retentionResolution5m }}"
-        - "--retention.resolution-1h={{ .Values.compact.retentionResolution1h }}"
-        - "--block-sync-concurrency={{ .Values.compact.blockSyncConcurrency }}"
-        - "--compact.concurrency={{ .Values.compact.compactConcurrency }}"
+        - "--consistency-delay={{ $root.Values.compact.consistencyDelay }}"
+        - "--retention.resolution-raw={{ $root.Values.compact.retentionResolutionRaw }}"
+        - "--retention.resolution-5m={{ $root.Values.compact.retentionResolution5m }}"
+        - "--retention.resolution-1h={{ $root.Values.compact.retentionResolution1h }}"
+        - "--block-sync-concurrency={{ $root.Values.compact.blockSyncConcurrency }}"
+        - "--compact.concurrency={{ $root.Values.compact.compactConcurrency }}"
         - "--wait"
-{{ with .Values.compact.extraArgs }}{{ toYaml . | indent 8 }}{{- end }}
+        {{- if $hashPartioning }}
+        - |
+          --selector.relabel-config=
+            - action: hashmod
+              source_labels: ["__block_id"]
```

Comment:
After some investigation, looks like this configuration doesn't work. I am thinking about making this source label configurable in the helm chart.

This label should identify the blocks from different clusters and group them together.

Reply:
ok